crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -220,8 +221,21 @@ impl Analyzer<'_, '_> {
                     }
 
                     Type::StringMapping(i) => {
+                        let ctx = Ctx {
+                            in_actual_type: opts.in_type_or_type_param,
+                            ..self.ctx
+                        };
+
                         let ty = self
-                            .expand_intrinsic_types(actual_span, i)
+                            .with_ctx(ctx)
+                            .expand_intrinsic_types(
+                                actual_span,
+                                i,
+                                AssignOpts {
```

Comment:
I think we should pass `span`, not `AssignOpts`

Reply:
Logically the function for expanding accepting options are assignment is strange